mllib/src/main/scala/org/apache/spark/mllib/linalg/distributed/BlockMatrix.scala

```diff
@@ -172,6 +174,36 @@ class BlockMatrix(
     assert(cols <= nCols, s"The number of columns $cols is more than claimed $nCols.")
   }
 
+  def validate(): Unit = {
+    logDebug("Validating BlockMatrix...")
+    // check if the matrix is larger than the claimed dimensions
+    estimateDim()
+    logDebug("BlockMatrix dimensions are okay...")
+
+    // Check if there are multiple MatrixBlocks with the same index.
+    val indexCounts = blockInfo.countByKey().foreach { case (key, cnt) =>
+      if (cnt > 1) {
+        throw new SparkException(s"There are MatrixBlocks with duplicate indices. Please remove " +
+          s"blocks with duplicate indices. You may call reduceByKey on the underlying RDD and " +
+          s"sum the duplicates. You may convert the matrices to Breeze before summing them up.")
```

Comment:
You have `s"..."` but no variable is used inside. We should at least put `key` to the error message. For the error message, it should be sufficient to say "Found duplicate block coordinate: $key.". `sum the duplicates` is just our guess, which may confuse users if they should do something else.